### PR TITLE
Update rusty_v8 to 0.26.0

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Related to #119 

This pr updates rusty_v8 and (will fix) ~fixes~ the broken build and tests.

### CODE REVIEW

**Comment by Ethan-Arrowood** [comment]
Looks like the two failing tests are in the CLI. It seems like stderr isn't being propagated correctly. Unfortunately rusty_v8 doesn't have any release notes so I'm not really sure what changed since the last version, but I'm digging into their unit tests now to see what I can figure out. 

(Also this is my first rust based open source contribution; any tips welcome!)

**Comment by devsnek** [comment]
I started to work on this (though feel free to take over!), basically module evaluation returns a promise that jstime needs to wait for (or really just unwrap, since there's no event loop yet)

**Comment by Ethan-Arrowood** [comment]
> I started to work on this (though feel free to take over!) 

Thank you!

> module evaluation returns a promise that jstime needs to wait for (or really just unwrap, since there's no event loop yet)

Okay great I'll look into this

**Comment by Ethan-Arrowood** [comment]
Am I looking at the right code block? 

https://github.com/jstime/jstime/blob/7734ba9fe5f9bf35f8bdad3eb7e2728e86e67923/core/src/module.rs#L40-L56

I see in the rusty_v8 docs for [`m.evaluate`](https://docs.rs/rusty_v8/0.26.0/rusty_v8/struct.Module.html#method.evaluate) that when it fails the error is available via `|GetException|`. Additionally since `scope` is a `TryCatch` I was looking at [this test case](https://github.com/denoland/rusty_v8/blob/c525555e4291de8b3a2357576d11dcb2c049c69d/tests/test_api.rs#L687-L700) in rusty.

**Comment by devsnek** [comment]
The docs appear to be outdated, and i think the tests coincidentally pass. if you updated the test to `await 1; throw new Error('foo');` it would fail.

**Comment by Ethan-Arrowood** [comment]
Hmm is it the `m.evaluate` call? 

I added:

```rust
                let res = m.evaluate(scope).unwrap();
                println!("{:?}", scope.has_caught());
                println!("{:?}", scope.stack_trace());
                Ok(res)
```
But it prints `false` and `None` when I run `cargo run "./cli/tests/fixtures/function-does-not-exist.js"`

**Comment by devsnek** [comment]
i think `res` there should be a promise, you should be able to check if its rejected (a hack, relying on our lack of event loop) and extract the rejection value from it.

**Comment by Ethan-Arrowood** [comment]
Awesome, 

```rust
let res = m.evaluate(scope).unwrap();
println!("{:?}", v8::Value::is_promise(&res));
```
Prints `true` 

Getting there!

**Comment by Ethan-Arrowood** [comment]
This might be my novice-Rust skills showing but how do I get `&res` to be a `Promise`? Its saying it is a `Local`.

My (naive) TypeScript brain wants to do something like this:
```rust
if v8::Value::is_promise(&res) {
  println!("{:?}", v8::Promise::result(&res, scope));
}
```

**Comment by Ethan-Arrowood** [comment]
Hmm maybe:

```rust
let res = m.evaluate(scope).unwrap();
let p = unsafe { v8::Local::<v8::Promise>::cast(res) };
```

**Comment by Ethan-Arrowood** [comment]
Well I got the test cases to pass 🎉 Excited to see how to improve this mess of a code block 😂

File: core/src/module.rs [hunk 1]
```diff
@@ -48,8 +48,12 @@ impl Loader {
             Some(m) => {
                 m.instantiate_module(scope, module_resolve_callback)
                     .unwrap();
-                m.evaluate(scope)
-                    .map_or_else(|| Err(scope.stack_trace().unwrap()), Ok)
+                let res = m.evaluate(scope).unwrap();
+                let p = unsafe { v8::Local::<v8::Promise>::cast(res) };
+                match p.state() {
+                    v8::PromiseState::Fulfilled => Ok(p.result(scope)),
+                    v8::PromiseState::Rejected | v8::PromiseState::Pending => Err(p.result(scope)),
```
Comment (devsnek) [inline core/src/module.rs:55]: pending state should be panic, there's no result to return and it should be unreachable until there's an event loop.

**Comment by Ethan-Arrowood** [comment]
Will do!

File: cli/tests/fixtures/module-graph-exec-order/a.mjs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  console.log('promise a')
+  queueMicrotask(() => {
+    console.log('microtask a')
+    resolve('a')
+  })
+})
+
+console.log('end a')
+
+export default a
```
Comment (MylesBorins) [inline cli/tests/fixtures/module-graph-exec-order/a.mjs:17]:
```suggestion
export default a

```

File: cli/tests/fixtures/module-graph-exec-order/b.mjs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  console.log('promise b')
+  queueMicrotask(() => {
+    console.log('microtask b')
+    resolve('b')
+  })
+})
+
+console.log('end b')
+
+export default b
```
Comment (MylesBorins) [inline cli/tests/fixtures/module-graph-exec-order/b.mjs:13]:
```suggestion
export default b

```

File: cli/tests/fixtures/module-graph-exec-order/c.mjs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  console.log('promise c')
+  queueMicrotask(() => {
+    console.log('microtask c')
+    resolve('c')
+  })
+})
+
+console.log('end c')
+
+export default c
```
Comment (MylesBorins) [inline cli/tests/fixtures/module-graph-exec-order/c.mjs:13]:
```suggestion
export default c

```

File: cli/tests/fixtures/module-graph-exec-order/root.mjs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import a from './a.mjs'
+import c from './c.mjs'
+
+console.log('start root')
+
+console.log(a)
+console.log(c)
+
+console.log('end root')
```
Comment (MylesBorins) [inline cli/tests/fixtures/module-graph-exec-order/root.mjs:9]:
```suggestion
console.log('end root')

```

**Review by MylesBorins — APPROVED** [review]
LGTM

**Comment by MylesBorins** [comment]
LGTM

Would like to see someone else with a bit more rust experience throw an LGTM as well. Maybe @devsnek or @bengl

If folks don't have time to review by mid-week I'll go ahead and land this with just my LGMT

**Review by devsnek — APPROVED** [review]

**Comment by Ethan-Arrowood** [comment]
🎉 yay this was fun to work on. Looking forward to doing more rust in the future!

**Pull request merged**